crates/bevy_ecs/src/bundle.rs

```diff
@@ -0,0 +1,233 @@
+pub use bevy_ecs_macros::Bundle;
+
+use crate::{
+    component::{Component, ComponentFlags, ComponentId, Components, StorageType, TypeInfo},
+    entity::Entity,
+    storage::{SparseSetIndex, SparseSets, Table},
+};
+use bevy_ecs_macros::all_tuples;
+use std::{any::TypeId, collections::HashMap};
+
+/// A dynamically typed ordered collection of components
+///
+/// See [Bundle]
+pub trait DynamicBundle: Send + Sync + 'static {
+    /// Gets this [DynamicBundle]'s components type info, in the order of this bundle's Components
+    fn type_info(&self) -> Vec<TypeInfo>;
+
+    /// Calls `func` on each value, in the order of this bundle's Components
+    #[doc(hidden)]
```

Comment:
Why is this marked as hidden?

Reply:
carry over from the old impl. this can be removed